docker-compose.yml

```diff
@@ -3,18 +3,61 @@
 version: '3'
 
 services:
+
```

Comment:
Generally looks good, but I think some of these are not necessary and we can tighten up the file, sorting, and naming conventions to something like this:

```
  # Define a service for the database.
  db:
    image: postgres:14-alpine
    # Docs for options to the postgres server command:
    # https://www.postgresql.org/docs/current/app-postgres.html
    command: postgres -c "log_lock_waits=on" -N 1000 -c "fsync=off"
    # Load environment variables for local development.
    env_file: ./app/local.env
    # Fix running on Apple silicon.
    platform: linux/amd64
    volumes:
      - db_data:/var/lib/postgresql/data

  # Define a service for the application.
  app:
    # This service relies on building from a Dockerfile in this file directory,
    # not on an image repository.
    build: ./app
    # Load environment variables for local development.
    env_file: ./app/local.env
    # NOTE: These values take precendence if the same value is specified in the env_file.
    environment:
      # The env_file defines DB_HOST=localhost for accessing a non-dockerized database.
      # In the docker-compose, we tell the app to use the dockerized database service instead.
      - DB_HOST=db
    # Fix running on Apple silicon.
    platform: linux/amd64
    # Expose the application port for local development.
    ports:
      - 8080:8080
    # Mount application directory for local development.
    volumes:
      - ./app:/srv

volumes:
  # Define a named volume to persist database data.
  db_data:
```

Reply:
Yeah, turns out a lot of the fluff with networks isn't needed, trimmed that out, but did leave the depends_on as it will make the app start up the DB automatically.

Reply:
Sounds good to me. Generally, `depends_on` is only useful if you are using `docker-compose up main-app`, but since that's a possibility for our use case, I'm good with keeping it!

I added/edited some comments into the suggestion above. Could you incorporate those into your changes?

Reply:
I think I have everything the same here, am I missing a diff?

Reply:
I added the changes as suggestions.